avro-fastserde/src/main/java/com/linkedin/avro/fastserde/PrimitiveFloatList.java

```diff
@@ -25,23 +31,33 @@
  * - It re-implements {@link #compareTo(GenericArray)}, {@link #equals(Object)} and {@link #hashCode()}
  *   in order to leverage the primitive types, rather than causing unintended boxing.
  *
- * TODO: Provide arrays for other primitive types.
+ *   Using ByteBuffer to speed up float-array deserialization: We allocate ByteBuffer to store the raw bytes from
+ *   BinaryDecoder and deserialize them only during array element access. We cache the results into the elements array
+ *   after the first get access of the array so that sub-sequent array access are fast.
+ *
+ *   TODO: Provide arrays for other primitive types.
  */
 public class PrimitiveFloatList extends AbstractList<Float>
     implements GenericArray<Float>, Comparable<GenericArray<Float>> {
   private static final float[] EMPTY = new float[0];
+  private static final int FLOAT_SIZE = Float.BYTES;
   private static final Schema FLOAT_SCHEMA = Schema.create(Schema.Type.FLOAT);
   private static final Schema SCHEMA = Schema.createArray(FLOAT_SCHEMA);
-
   private int size;
   private float[] elements = EMPTY;
```

Comment:
Is there any advantage to caching? How about making deserialization lazy every single time the array is accessed? Can we run a benchmark where we access the array more than once, and try to assess if the caching brings a benefit? It would be nice if we could avoid instantiating this array.

Reply:
That is a good question.
@majisourav99 
Could you do a quick benchmark?
Without caching, the complexity here will be reduced greatly.

Reply:
I tried reading the array twice and caching vs non-caching gives around ~20% better performance. But caching is here mostly to support the full arrayList interface of add/remove functionality.